core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -1148,6 +1148,34 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.ENFORCE)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey MASTER_AUTO_MOUNT_ENABLED =
+      new Builder(Name.MASTER_AUTO_MOUNT_ENABLED)
+          .setDescription("If enabled, Alluxio will attempt to mount UFS for foreign URIs.")
+          .setDefaultValue(false)
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
+          .setScope(Scope.MASTER)
+          .build();
+  public static final PropertyKey MASTER_AUTO_MOUNT_ROOT =
+      new Builder(Name.MASTER_AUTO_MOUNT_ROOT)
+          .setDescription("Alluxio root path for auto-mounted UFSes.")
+          .setDefaultValue("/auto-mount")
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
+          .setScope(Scope.MASTER)
+          .build();
+  public static final PropertyKey MASTER_AUTO_MOUNT_READONLY =
+      new Builder(Name.MASTER_AUTO_MOUNT_READONLY)
+          .setDescription("If true, auto-mounts UFSes as read-only.")
+          .setDefaultValue(true)
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
+          .setScope(Scope.MASTER)
+          .build();
+  public static final PropertyKey MASTER_AUTO_MOUNT_SHARED =
+      new Builder(Name.MASTER_AUTO_MOUNT_SHARED)
+          .setDescription("If true, auto-mounts UFSes as not-shared.")
```

Comment:
NOT shared? that seems confusing, since the key is called `alluxio.master.auto.mount.shared`.